react-components/src/util/Autocomplete.js

```diff
@@ -1,107 +1,244 @@
-    } = props;
-
-    let SelectComponent = null;
-
-    switch (variant) {
-        case 'creatable':
-            SelectComponent = Select.Creatable;
-            break;
-        case 'async':
-            SelectComponent = Select.Async;
-            break;
-        case 'asyncCreatable':
-            SelectComponent = Select.AsyncCreatable;
-            break;
-        default:
-            SelectComponent = Select;
-            break;
+
+    getOptionLabel(option) {
+        let labelKey = this.props.labelKey;
+        return option[labelKey]
     }
 
-    return (
-        <SelectComponent optionComponent={Option}
-                   {...custom} />
-    )
-}
+    getOptionValue(option) {
+        let valueKey = this.props.valueKey;
+        return option[valueKey]
+    }
 
-class Option extends Component {
-    handleClick = event => {
-        this.props.onSelect(this.props.option, event);
-    };
+    //This should hopefully be not needed in v2.1 for Creatable
+    isValidNewOption(inputValue, selectValue, selectOptions) {
+        let labelKey = this.props.labelKey;
+        if (inputValue.trim().length === 0 || selectOptions.find(option => option[labelKey] === inputValue)) {
+            return false;
+        }
+        return true;
+    }
+
+    //This should hopefully be not needed in v2.1 for Creatable
+    getNewOptionData(inputValue, optionLabel) {
+        let labelKey = this.props.labelKey;
+        return {
+            id: inputValue,
+            [labelKey]: optionLabel
+        }
+    }
 
     render() {
-        const {children} = this.props;
+        let {
+            variant,
+            classes,
+            controlShouldRenderValue,
+            CustomOption,
+            CustomControl,
+            CustomNoOptionsMessage,
+            CustomPlaceholder,
+            CustomSingleValue,
+            CustomMultiValue,
+            CustomValueContainer,
+            ...custom
+        } = this.props;
+
+        let SelectComponent = null;
+
+        switch (variant) {
+            case 'creatable':
+                SelectComponent = Creatable;
+                break;
+            case 'async':
+                SelectComponent = Async;
+                break;
+            case 'asyncCreatable':
+                SelectComponent = AsyncCreatableSelect;
+                break;
+            default:
+                SelectComponent = Select;
+                break;
+        }
+
+        let components = {
+            Option: CustomOption,
+            Control: CustomControl,
+            NoOptionsMessage: CustomNoOptionsMessage,
+            Placeholder: CustomPlaceholder,
+            SingleValue: CustomSingleValue,
+            MultiValue: CustomMultiValue,
+            ValueContainer: CustomValueContainer,
+        };
 
         return (
-            <MenuItem onClick={this.handleClick}>
-                {children}
-            </MenuItem>
-        );
+            <div className={classes.root}>
+                <NoSsr>
```

Comment:
Do we need this tag ? We don't use server side rendering as far as I know...

Reply:
Ah, good catch.  Left behind from some copy-paste 😊